duviz.py

```diff
@@ -246,6 +246,205 @@ def bar(self, label: str, width: int, fill='-', left='[', right=']', small='|',
         return b
 
 
+class ColumnsRenderer(TreeRenderer):
+    """
+    Render a SizeTree with vertical columns.
+    """
+
+    def __init__(
+        self,
+        height: int,
+        max_depth: int = 5,
+        size_formatter: SizeFormatter = SIZE_FORMATTER_COUNT,
+        color_mode: bool = False,
+    ):
+        super().__init__(max_depth, size_formatter)
+        self.height = height
+        self.color_mode = color_mode
+        if color_mode:
+            self.colorizers = []
+            c = Colorizer()
+            # We use reverse order so that the root is always the same color regardless of the depth
+            for i in reversed(range(max_depth + 1)):
+                cycle = itertools.cycle(
+                    c.COLOR_CYCLE_BMC if i % 2 == 0 else c.COLOR_CYCLE_RGY
+                )
+                self.colorizers.append(cycle)
+
+    def render(self, tree: SizeTree, width: int) -> List[str]:
+        lines = self._render(
+            tree, width // (self.max_depth + 1), self.height, self.max_depth
+        )
+        if lines and not self.color_mode:
+            # The monochrome rendering must be displayed in reverse mode
+            lines[0] = "\x1b[7m" + lines[0]
+            lines[-1] += "\x1b[27m"
+        return lines
+
+    def render_node(
+        self,
+        node: Optional[SizeTree],
+        width: int,
+        height: int,
+        depth: int,
+    ) -> List[str]:
+        """Render a single node"""
+        def _truncate(s, maxlen):
+            truncation_indicator = "..."
+            if len(s) <= maxlen:
+                return s
+            return s[: maxlen - len(truncation_indicator)] + truncation_indicator
+
+        if self.color_mode:
+            color = next(self.colorizers[depth])
+            prefix = color
+            suffix = Colorizer.COLOR_RESET
+            left = ""
+            border_lines = 0
+        else:
+            prefix = ""
+            suffix = ""
+            left = "█"
```

Comment:
Why this special character? can't this just be done with pure ASCII, or even just a space?

Reply:
A space could also work as a separator, but it would need to be inverted. While the "█" character is outside of seven-bit ASCII, it is present in all monospace fonts I encountered, even in Linux's text-mode console, probably because it was a part of CP437 of the original IBM PC.

On the other hand, I realized that the "▁" character (which was also used in this revision) is missing from CP437 and has no corresponding glyph in Linux's text-mode console either. For this reason I have been working on a new revision that produces a less attractive but more compatible output that looks like this:

![duviz-columns-ascii](https://user-images.githubusercontent.com/20930533/196029420-35002835-d318-483a-b8e1-cbce217210be.png)

I still need to do some finishing touches before I can upload this new revision though.